client/allocrunner/taskrunner/plugin_supervisor_hook.go

```diff
@@ -335,10 +335,10 @@ func (h *csiPluginSupervisorHook) supervisorLoopOnce(ctx context.Context, socket
 	}
 
 	client, err := csi.NewClient(socketPath, h.logger.Named("csi_client").With("plugin.name", h.task.CSIPluginConfig.ID, "plugin.type", h.task.CSIPluginConfig.Type))
-	defer client.Close()
 	if err != nil {
 		return false, fmt.Errorf("failed to create csi client: %v", err)
 	}
+	defer client.Close()
```

Comment:
This fixes a panic that can't happen in 0.11.2 or earlier because we never actually connect.